lib/compute/jenkins-main-node.ts

```diff
@@ -288,8 +298,40 @@ export class JenkinsMainNode {
 
       // install all the list of plugins from the list and restart (done in same command as restart is to be done after completion of install-plugin)
       // eslint-disable-next-line max-len
-      InitCommand.shellCommand(`java -jar jenkins-cli.jar -s http://localhost:8080 -auth @/var/lib/jenkins/secrets/myIdPassDefault install-plugin ${JenkinsPlugins.plugins.join(' ')} && java -jar jenkins-cli.jar -s http://localhost:8080 -auth @/var/lib/jenkins/secrets/myIdPassDefault restart`),
+      InitCommand.shellCommand(`java -jar /jenkins-cli.jar -s http://localhost:8080 -auth @/var/lib/jenkins/secrets/myIdPassDefault install-plugin ${JenkinsPlugins.plugins.join(' ')} && java -jar jenkins-cli.jar -s http://localhost:8080 -auth @/var/lib/jenkins/secrets/myIdPassDefault restart`),
       // Warning : any commands after this may be executed before the above command is complete
+
+      // Commands are fired one after the other but it does not wait for the command to complete.
+      // Therefore, sleep 60 seconds to wait for plugins to install and jenkins to start which is required for the next step
+      InitCommand.shellCommand('sleep 60'),
+
+      // If devMode is false, first line extracts the oidcFederateProps as json from the secret manager
+      // xmlstarlet is used to setup the securityRealm values for oidc
+      InitCommand.shellCommand(oidcFederateProps.devMode ? 'echo Not altering the jenkins config.xml in dev-mode'
+        // eslint-disable-next-line max-len
+        : `var=\`aws --region ${stackRegion} secretsmanager get-secret-value --secret-id ${oidcFederateProps.oidcCredArn} --query SecretString --output text\` &&`
+        + 'xmlstarlet ed -L -d "/hudson/securityRealm" \\'
```

Comment:
Its really hard to know if this is correct and if you make a change if that would ruin everything.  Lets move this into a separate class that can produce this string for feeding into xmlstarlet, it might be worthwhile to even make this into a python script and just execute that

I see three areas of focus:
1. Finding/replacing the security configuration
2. Retrieving the secret
3. Creating a correctly formed security configuration